ui/src/app/components/rule/rule.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit, SimpleChanges } from '@angular/core'
+import { SidenavService } from 'src/app/services/sidenav/sidenav.service'
+import { DataService } from 'src/app/services/data/data.service'
+import IRuleContent from 'src/app/model/Rule'
+
+@Component({
+  selector: 'app-rule',
+  templateUrl: './rule.component.html',
+  styleUrls: ['./rule.component.scss'],
+})
+export class RuleComponent implements OnInit {
+  dataSource: IRuleContent[] = []
+  displayedColumns = ['order', 'name', 'type', 'objectType', 'associatedObject', 'enabled']
+
+  constructor(private sidenavService: SidenavService, private data: DataService) {}
+
+  ngOnInit(): void {
+    this.data.rule.subscribe({
+      next: (data: IRuleContent) => {
+        const newData = [...this.dataSource, data]
+        this.dataSource = newData
+      },
+    })
+  }
+  ngOnChanges(changes: SimpleChanges): void {
```

Comment:
Rule not getting updated here after adding it via add rule button

Reply:
An issue logged and we are working on it.